src/lib/TurboConfirm.js

```diff
@@ -0,0 +1,99 @@
+import ConfirmationController from './ConfirmationController'
+
+export class TurboConfirm {
+  #controller
+  #config = {
+    dialogSelector: '#confirm',
+    activeClass: 'modal--active',
+    acceptSelector: '#confirm-accept',
+    denySelector: '.confirm-cancel',
+    animationDuration: 300,
+    showConfirmCallback: element => element.showModal && element.showModal(),
+    hideConfirmCallback: element => element.close && element.close(),
+    messageSlotSelector: '#confirm-title',
+    contentSlots: {
+      body: {
+        contentAttribute: 'confirm-details',
+        slotSelector: '#confirm-body'
+      },
+      acceptText: {
+        contentAttribute: 'confirm-button',
+        slotSelector: '#confirm-accept'
+      }
+    }
+  }
+
+  constructor(options = {}) {
+    for(const [key, value] of Object.entries(options)) {
+      this.#config[key] = value
+    }
+
+    this.#controller = new ConfirmationController(this.#config.dialogSelector, this)
+  }
+
+  confirm(message, _formElement, submitter) {
+    const clickTarget = this.#clickTarget(submitter)
+    const contentMap = this.#contentMap(message, clickTarget)
+
+    return this.confirmWithContent(contentMap)
+  }
+
+  confirmWithContent(contentMap) {
+    return new Promise(resolve => this.#controller.showConfirm(contentMap, resolve))
```

Comment:
Out of curiosity, is there a reason you're building the promise here instead of in `ConfirmationController.js`?

Reply:
@justwiebe. I never tried it the other way, but I like that the promise is visible from both sides.  Do you think it would be cleaner if the Promise were created in ConfirmationController?  It would still be returned here.

Reply:
@justwiebe thanks for the feedback, I'll experiment with this after work

Reply:
@justwiebe I now concur, your suggestion is indeed an improvement.  😆  Thank you!

This refactor prompted several others as well.  Also added JSDoc comments for the 2 public APIs.